# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/cookie/ThirdPartyCookieBlockingExceptionListService.sys.mjs
# Commit: 336214ad7891
# Full Hash: 336214ad789156999085eb7607fe6a6cbf7bbfe9
# Author: Tim Huang <tihuang@mozilla.com>
# Date: 2025-01-10 21:44:02
# Regressor Bug: 1915355
# File Overlap Count: 1
# Description:
#   Bug 1915355 - Part 4: Implement the third-party cookie blocking excpetion in CookieService. r=cookie-reviewers,baku,valentin
#   
#   The patch integrates the third-party cookie blocking excpetion in
#   CookieService.
#   
# ==============================================================================

diff -r 453324c772ae -r 336214ad7891 netwerk/cookie/ThirdPartyCookieBlockingExceptionListService.sys.mjs
--- a/netwerk/cookie/ThirdPartyCookieBlockingExceptionListService.sys.mjs	Fri Jan 10 08:24:29 2025 +0000
+++ b/netwerk/cookie/ThirdPartyCookieBlockingExceptionListService.sys.mjs	Fri Jan 10 08:24:29 2025 +0000
@@ -57,10 +57,10 @@
 
   #handleExceptionChange(created = [], deleted = []) {
     if (created.length) {
-      // TODO: Calling CookieService API to remove exception sites.
+      Services.cookies.addThirdPartyCookieBlockingExceptions(created);
     }
     if (deleted.length) {
-      // TODO: Calling CookieService API to add exception sites.
+      Services.cookies.removeThirdPartyCookieBlockingExceptions(deleted);
     }
   }
 